### ISSUE TITLE
test_SEH() of test_ctypes logs "Windows fatal exception: access violation"

### ISSUE THREAD

**Issue by vstinner** [python/cpython#74311]
BPO | [30125](https://bugs.python.org/issue30125)
--- | :---
Nosy | @pfmoore, @vstinner, @tjguk, @zware, @zooba
PRs | <li>python/cpython#1237</li><li>python/cpython#1240</li><li>python/cpython#1243</li><li>python/cpython#1343</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-05-02.14:35:25.336>
created_at = <Date 2017-04-21.12:23:18.666>
labels = ['3.7', 'OS-windows']
title = 'test_SEH() of test_ctypes logs "Windows fatal exception: access violation"'
updated_at = <Date 2017-05-02.14:35:25.334>
user = 'https://github.com/vstinner'
```

bugs.python.org fields:
```python
activity = <Date 2017-05-02.14:35:25.334>
actor = 'vstinner'
assignee = 'none'
closed = True
closed_date = <Date 2017-05-02.14:35:25.336>
closer = 'vstinner'
components = ['Windows']
creation = <Date 2017-04-21.12:23:18.666>
creator = 'vstinner'
dependencies = []
files = []
hgrepos = []
issue_num = 30125
keywords = []
message_count = 6.0
messages = ['292030', '292057', '292078', '292081', '292533', '292765']
nosy_count = 5.0
nosy_names = ['paul.moore', 'vstinner', 'tim.golden', 'zach.ware', 'steve.dower']
pr_nums = ['1237', '1240', '1243', '1343']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue30125'
versions = ['Python 3.7']
```

</p></details>

**Comment by vstinner** [comment]
Attached PR removes the following scary log.

http://buildbot.python.org/all/builders/AMD64%20Windows8.1%20Non-Debug%203.x/builds/655/steps/test/logs/stdio

Windows fatal exception: access violation

Current thread 0x00000b88 (most recent call first):
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\unittest\\case.py", line 178 in handle
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\unittest\\case.py", line 733 in assertRaises
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\ctypes\\test\\test_win32.py", line 47 in test_SEH
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\unittest\\case.py", line 605 in run
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\unittest\\case.py", line 653 in \_\_call__
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\unittest\\suite.py", line 122 in run
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\unittest\\suite.py", line 84 in \_\_call__
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\unittest\\suite.py", line 122 in run
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\unittest\\suite.py", line 84 in \_\_call__
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\unittest\\suite.py", line 122 in run
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\unittest\\suite.py", line 84 in \_\_call__
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\unittest\\suite.py", line 122 in run
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\unittest\\suite.py", line 84 in \_\_call__
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\unittest\\suite.py", line 122 in run
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\unittest\\suite.py", line 84 in \_\_call__
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\unittest\\runner.py", line 176 in run
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\test\\support\\__init__.py", line 1896 in \_run_suite
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\test\\support\\__init__.py", line 1930 in run_unittest
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\test\\libregrtest\\runtest.py", line 164 in test_runner
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\test\\libregrtest\\runtest.py", line 165 in runtest_inner
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\test\\libregrtest\\runtest.py", line 119 in runtest
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\test\\libregrtest\\runtest_mp.py", line 71 in run_tests_slave
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\test\\libregrtest\\main.py", line 470 in \_main
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\test\\libregrtest\\main.py", line 463 in main
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\test\\libregrtest\\main.py", line 527 in main
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\test\\regrtest.py", line 46 in \_main
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\test\\regrtest.py", line 50 in \<module\>
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\runpy.py", line 85 in \_run_code
  File "D:\\buildarea\\3.x.ware-win81-release\\build\\lib\\runpy.py", line 193 in \_run_module_as_main

**Comment by vstinner** [comment]
New changeset 46c2b81026bbf966c0898a1fa30d98c33673aea0 by Victor Stinner in branch 'master':
bpo-30125: Fix faulthandler.disable() on Windows (bpo-1240)
https://github.com/python/cpython/commit/46c2b81026bbf966c0898a1fa30d98c33673aea0

**Comment by vstinner** [comment]
New changeset 3a8f8ea2aca46f9ef1e715f2c924357b6719525f by Victor Stinner in branch '3.6':
bpo-30125: Fix faulthandler.disable() on Windows (bpo-1243)
https://github.com/python/cpython/commit/3a8f8ea2aca46f9ef1e715f2c924357b6719525f

**Comment by vstinner** [comment]
New changeset a36e939aeb3b5a2c56561eb24f0e339eee9f3f9d by Victor Stinner in branch 'master':
bpo-30125: disable faulthandler in ctypes test_SEH (bpo-1237)
https://github.com/python/cpython/commit/a36e939aeb3b5a2c56561eb24f0e339eee9f3f9d

**Comment by vstinner** [comment]
New changeset e005dd9a6da7dfa34f47a6989a7e0a6b1ad04949 by Victor Stinner in branch '3.6':
bpo-30125: disable faulthandler in ctypes test_SEH (bpo-1237) (bpo-1343)
https://github.com/python/cpython/commit/e005dd9a6da7dfa34f47a6989a7e0a6b1ad04949

**Comment by vstinner** [comment]
faulthandler.disable() has been fixed on Windows in Python 3.6 and master (3.7).

The test_ctypes warning has been made quiet in Python 3.6 and master (3.7) as well.
